modules/openapi-generator/src/main/resources/go-server/controller-api.mustache

```diff
@@ -344,13 +373,44 @@ func (c *{{classname}}Controller) {{nickname}}(w http.ResponseWriter, r *http.Re
 	{{^isBoolean}}
 	{{^isArray}}
 	{{#defaultValue}}
-	{{paramName}}Param := "{{defaultValue}}"
 	if query.Has("{{baseName}}") {
-		{{paramName}}Param = query.Get("{{baseName}}")
+		{{paramName}}Param := {{^isString}}{{dataType}}( {{/isString}}query.Get("{{baseName}}"){{^isString}} ){{/isString}}
 	}
```

Comment:
Is this an intended change?

seems like no one can now read from `{{paramName}}Param` since it is in the `if` blacok scope

Also the default value is removed

Reply:
Definitely not intended, great catch, thank you so much